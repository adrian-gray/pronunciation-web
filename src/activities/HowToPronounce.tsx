import React from "react";

import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";

import SplitHilite from "./../components/SplitHilite";

export default (props: {
  arr: string[];
  ipa: string;
  image: { url: string; alt: string; title: string };
}) => {
  const { arr, image, ipa } = props;
  const { url, alt, title } = image;

  const list = arr.map((el, i) => (
    <li key={i} className="list-group-item">
      {el}
    </li>
  ));

  return (
    <div className="headspace">
      <h3>
        {"How to pronounce "}
        <SplitHilite str={ipa} />
      </h3>

      <div className="sub-canvas">
        <Row>
          <Col>
            <div className="padded">
              <h5>{title}</h5>
            </div>
          </Col>
          <Col className="align-self-center">
            <ul className="list-group">{list}</ul>
          </Col>
        </Row>
      </div>
    </div>
  );
};
