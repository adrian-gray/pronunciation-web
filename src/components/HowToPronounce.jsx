import React from 'react'

import {
  Grid,
  Paper,
  List,
  ListItem,
  ListItemText,
  Typography,
  withStyles
} from '@material-ui/core'

import SplitHilite from './SplitHilite'

const styles = (theme) => ({
  headspace: theme.headspace,
  fullWidth: theme.fullWidth,
  subCanvas: theme.subCanvas
})

const HowToPronounce = (props) => {
  const { arr, classes, ipa, url } = props

  const list = arr.map((el, i) => (
    <ListItem key={i}>
      <ListItemText>
        {el}
      </ListItemText>
    </ListItem>
  ))

  return (
    <div className={classes.headspace}>
      <Typography variant='title' gutterBottom>
        {'How to pronounce '}<SplitHilite str={ipa} />
      </Typography>

      <Paper className={classes.subCanvas}>
        <Grid container spacing={24}>
          <Grid item xs={12} sm={6}>
            <img className={classes.fullWidth} src={url} />
          </Grid>
          <Grid item xs={12} sm={6}>
            <List>
              {list}
            </List>
          </Grid>
        </Grid>
      </Paper>
    </div>
  )
}

export default withStyles(styles)(HowToPronounce)
