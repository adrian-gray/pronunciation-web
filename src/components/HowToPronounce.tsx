import React from "react";

import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";

import SplitHilite from "./SplitHilite";

export default props => {
  const { arr, image, ipa } = props;
  const { url, alt, title } = image;

  const list = arr.map((el, i) => <li key={i}>{el}</li>);

  return (
    <div className="headspace">
      <h3>
        {"How to pronounce "}
        <SplitHilite str={ipa} />
      </h3>

      <Container className="sub-canvas">
        <Row>
          <Col>
            <div className="padded">
              <img className="fill-width" src={url} title={title} alt={alt} />
              <h5>{title}</h5>
            </div>
          </Col>
          <Col>
            <ul>{list}</ul>
          </Col>
        </Row>
      </Container>
    </div>
  );
};
