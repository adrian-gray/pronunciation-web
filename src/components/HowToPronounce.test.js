import React from "react";
import { shallow } from "enzyme";
import { render, cleanup } from "@testing-library/react";

import HowToPronounce from "./HowToPronounce";

afterEach(cleanup);

test("HowToPronounce", async () => {
  const props = {
    ipa: "a:",
    arr: ["foo", "bar"],
    image: {
      url: "foo.com",
      alt: "image",
      title: "image"
    }
  };

  const container = shallow(<HowToPronounce {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});
