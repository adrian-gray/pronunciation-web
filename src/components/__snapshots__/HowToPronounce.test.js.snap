// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowToPronounce 1`] = `undefined`;
